core/templates/components/main-nav/main-nav.twig

```diff
@@ -0,0 +1,44 @@
+{%- if markup is not empty %}
+  {# If custom markup is provided, emit it as is #}
+  {{ markup }}
+{% else -%}
+  {# If no custom markup, generate markup based on structured data #}
+<nav {{- attributes }} class="su-main-nav {{ modifier_class }}" aria-label="Main">
+  <button class="su-main-nav__toggle" aria-expanded="false">{{ toggle_text }}</button>
+  {% if lv1 is not empty -%}
```

Comment:
Combine this with the markup variable to have just one region variable that accepts two different data types. The first is the raw html (string) and the second is the iterable object. eg:

```
        {% if links is iterable %}
          {% for link in links %}
           .. do code things.
          {% endfor %}
        {% else %}
          {{ links }}
        {% endif %}
```

Reply:
This is done. Will wait till I see JB and see if she has any thoughts about your above comment with the data structure (adding link attributes) etc.